host/src/server/api/v2/proof/mod.rs

```diff
@@ -79,29 +82,55 @@ async fn proof_handler(
     .await?;
 
     let proof_type = proof_request.proof_type;
-    let request_key = SingleProofRequestKey::new(
-        chain_id,
-        proof_request.block_number,
-        blockhash,
-        proof_request.proof_type,
-        proof_request.prover.to_string(),
-    )
-    .into();
-    let request_entity = SingleProofRequestEntity::new(
+    let request_key =
+        GuestInputRequestKey::new(chain_id, proof_request.block_number, blockhash).into();
+    let request_entity = GuestInputRequestEntity::new(
         proof_request.block_number,
         proof_request.l1_inclusion_block_number,
-        proof_request.network,
-        proof_request.l1_network,
+        proof_request.network.clone(),
+        proof_request.l1_network.clone(),
         proof_request.graffiti,
-        proof_request.prover,
-        proof_request.proof_type,
-        proof_request.blob_proof_type,
-        proof_request.prover_args,
+        proof_request.blob_proof_type.clone(),
+        proof_request.prover_args.clone(),
     )
     .into();
 
+    // deal with guest input first
     let result = crate::server::prove(&actor, request_key, request_entity).await;
-    Ok(to_v2_status(proof_type, result))
+    match result {
+        Ok(InPoolStatus::Success { proof }) => {
+            let request_key = SingleProofRequestKey::new(
+                chain_id,
+                proof_request.block_number,
+                blockhash,
+                proof_request.proof_type,
+                proof_request.prover.to_string(),
+            )
+            .into();
+            let mut prover_args = proof_request.prover_args.clone();
+            prover_args.insert(
+                "guest_input".to_string(),
+                serde_json::to_value(proof.proof)?,
+            );
+            let request_entity = SingleProofRequestEntity::new(
+                proof_request.block_number,
+                proof_request.l1_inclusion_block_number,
+                proof_request.network,
+                proof_request.l1_network,
+                proof_request.graffiti,
+                proof_request.prover,
+                proof_request.proof_type,
+                proof_request.blob_proof_type,
+                prover_args,
+            )
+            .into();
+
+            let result = crate::server::prove(&actor, request_key, request_entity).await;
+            Ok(to_v2_status(proof_type, result))
```

Comment:
How about moving the changes to [`server/handler.rs`](https://github.com/taikoxyz/raiko/blob/main/host/src/server/handler.rs#L19),  so that we can reuse the logic amount all HTTP interfaces?

Reply:
could be.
this is the demo of the idea, if it's ok, we can move forward.

Reply:
because v2 proof is deprecated, we don't need to optimize it.

For v3, so far only BatchProof supports this, let's keep the impl like this for now, and refine it after 1.7.0 release.